tests/test_torch.py

```diff
@@ -48,7 +48,7 @@ def test_PyTorch(device, n_parallel, batch, output_key):
     with Pipeline() as p:
         input = Unpack([torch.tensor([float(i)]) for i in range(100)])
 
-        block = BatchPipeline(2) if batch else nullcontext(p)
+        block = BatchPipeline(2) if batch else ExitStack()
```

Comment:
I can see no added value in this change. If you want to use `ExitStack`, I would do the following:

```python
with ExitStack() as stack:
    if batch:
        stack.enter_context(BatchPipeline(2))
    result = ...
```